core/client/fs/src/main/java/alluxio/client/file/FileSystemContext.java

```diff
@@ -231,17 +230,7 @@ private synchronized void initContext(ClientContext ctx,
         .setMasterInquireClient(masterInquireClient).build();
     mMetricsEnabled = getClusterConf().getBoolean(PropertyKey.USER_METRICS_COLLECTION_ENABLED);
     if (mMetricsEnabled) {
-      try {
-        InetSocketAddress masterAddr = masterInquireClient.getPrimaryRpcAddress();
-        mMasterClientContext.loadConf(masterAddr, true, true);
-      } catch (UnavailableException e) {
-        LOG.error("Failed to get master address during initialization", e);
-      } catch (AlluxioStatusException ae) {
-        LOG.error("Failed to load configuration from "
-            + "meta master during initialization", ae);
-      }
-      MetricsSystem.startSinks(getClusterConf().get(PropertyKey.METRICS_CONF_FILE));
-      MetricsHeartbeatContext.addHeartbeat(getClientContext(), masterInquireClient);
+      MetricsHeartbeatContext.asyncAddHeartbeat(getClientContext(), masterInquireClient);
```

Comment:
Is it necessary to start the metrics system asynchronously? Only the heartbeat needs the master address right? Instead of computing master address, could we compute it lazily during the heartbeat call (and thus use the heartbeat thread, making the submission of the heartbeat thread lightweight)?